src/components/Button.tsx

```diff
@@ -0,0 +1,176 @@
+    disabled: gray,
+    disabledLink: "none"
+  },
+  color: {
+    [ButtonType.PRIMARY]: white,
+    [ButtonType.SECONDARY]: white,
+    [ButtonType.LINK]: linkStyles.color,
+    disabled: white,
+    disabledLink: gray
+  },
+  size: {
+    [ButtonSize.SMALL]: ".8rem",
+    [ButtonSize.REGULAR]: "1rem",
+    [ButtonSize.LARGE]: "1.2rem"
+  },
+  padding: {
+    [ButtonSize.SMALL]: "5px 10px",
+    [ButtonSize.REGULAR]: "7.5px 15px",
+    [ButtonSize.LARGE]: "10px 20px"
+  },
+  textDecoration: {
+    [ButtonType.LINK]: linkStyles.textDecoration
+  },
+  hover: {
+    background: {
+      [ButtonType.PRIMARY]: interactive.primaryHover,
+      [ButtonType.SECONDARY]: interactive.secondaryHover,
+      [ButtonType.LINK]: "none"
+    },
+    color: {
+      [ButtonType.PRIMARY]: "white",
+      [ButtonType.SECONDARY]: "white",
+      [ButtonType.LINK]: linkStyles.hover.color
+    },
+    textDecoration: {
+      [ButtonType.LINK]: linkStyles.hover.textDecoration
+    }
+  }
+};
+
+interface ThemeProps {
+  type: ButtonType;
+  size: ButtonSize;
+}
+const styled = styledUntyped as ThemedBaseStyledInterface<ThemeProps>;
+const ButtonElem = styled.button`
+  cursor: ${props => (props.disabled ? "not-allowed" : "pointer")};
+  padding: ${props =>
+    props.theme.type === ButtonType.LINK
+      ? 0
+      : buttonStyles.padding[props.theme.size]};
+  border: none;
+  border-radius: 2px;
+
+  color: ${props =>
+    props.disabled
+      ? props.theme.type === ButtonType.LINK
+        ? buttonStyles.color.disabledLink
+        : buttonStyles.color.disabled
+      : buttonStyles.hover.color[props.theme.type]};
+  background: ${props =>
+    props.disabled
+      ? props.theme.type === ButtonType.LINK
+        ? buttonStyles.background.disabledLink
+        : buttonStyles.background.disabled
+      : buttonStyles.hover.background[props.theme.type]};
+  text-decoration: none;
+
+  font-size: ${props => buttonStyles.size[props.theme.size]};
+  font-weight: ${props =>
+    props.theme.type === ButtonType.LINK ? linkStyles.fontWeight : "500"};
+
+  transition: color 0.15s, text-decoration 0.15s, background-color 0.15s;
+
+  :hover,
+  :active,
+  :focus {
+    background: ${props =>
+      props.disabled
+        ? props.theme.type === ButtonType.LINK
+          ? buttonStyles.background.disabledLink
+          : buttonStyles.background.disabled
+        : buttonStyles.background[props.theme.type]};
+    color: ${props =>
+      props.disabled
+        ? props.theme.type === ButtonType.LINK
+          ? buttonStyles.color.disabledLink
+          : buttonStyles.color.disabled
+        : buttonStyles.color[props.theme.type]};
+    text-decoration: ${props =>
+      props.theme.type === ButtonType.LINK
+        ? buttonStyles.hover.textDecoration[props.theme.type]
+        : "none"};
+  }
+`;
+
+/**
+ * General purpose button that matches application styles.
+ * Not intended for webpage linking, but instead for running
+ * code on click.
```

Comment:
Could you add an example of how to use this component to this comment?

Reply:
8c2ec18